_resources/port1.0/group/crossgcc-1.0.tcl

```diff
@@ -77,12 +82,10 @@ proc crossgcc.setup {target version} {
         homepage        http://gcc.gnu.org/
         master_sites    gnu:gcc/gcc-${version}/:gcc
 
-        # TODO: cheating
-        # the xz is used for 6.4.0, 7.2.0 and probably later
-        if {[lindex [split ${crossgcc.version} .] 0] < 7} {
-            use_bzip2   yes
-        } else {
-            use_xz      yes
+        if {[info exists crossgcc.versions_info($version)]} {
+            use_[lindex [set crossgcc.versions_info($version)] 0] yes
+
+            checksums   {*}[lindex [set crossgcc.versions_info($version)] 1]
```

Comment:
Cool, I like this approach. For backward compatibility we probably still need the `else` part setting either `bzip2` when the version is less than 5 or 6 (and make sure that all versions since then are in the hash) or `xz` otherwise.

The next (less important) question arises: could these versions be shared with gcc ports? See also https://trac.macports.org/ticket/50904 about `distname`.